File: kata/almost_binary/almost_binary.py
import sys

SAMPLE_INPUT = """\
#.#.#.###.#.##.#
##.##.......#..#
#..#####..#.#...
###..#....###.##
#..#..#.#....##
#......#.##....
#############.#.""".splitlines()

SAMPLE_OUTPUT = """\
43949
55305
40744
58427
18755
16560
65530""".splitlines()


def solve(line: str) -> int:
    x = line.replace('#', '1').replace('.', '0')
    return int(x, 2)


def main():
    assert SAMPLE_OUTPUT == [str(solve(line)) for line in SAMPLE_INPUT]
    for line in sys.stdin:
        print(solve(line))


if __name__ == '__main__':
    main()

File: kata/almost_binary/almost_binary.rb
# Translated from Python to Ruby by ChatGPT

SAMPLE_INPUT = [
  '#.#.#.###.#.##.#',
  '##.##.......#..#',
  '#..#####..#.#...',
  '###..#....###.##',
  '#..#..#.#....##',
  '#......#.##....',
  '#############.#.'
]

SAMPLE_OUTPUT = %w[
  43949
  55305
  40744
  58427
  18755
  16560
  65530
]

def solve(line)
  x = line.gsub('#', '1').gsub('.', '0')
  x.to_i(2)
end

def main
  raise 'Assertion failed' unless SAMPLE_OUTPUT == SAMPLE_INPUT.map { |line| solve(line).to_s }

  ARGF.each_line do |line|
    puts solve(line.chomp)
  end
end

main
